models/activity.js

```diff
@@ -0,0 +1,56 @@
+// Import mongoose library
+const mongoose = require("mongoose");
+
+/* 
+Mongoose schema definition. See - https://mongoosejs.com/docs/guide.html#definition
+
+Defines the Activity.reference schema which consist of text and url. 
+
+Text: The text value is of type string, is required when the url value has characters of one or more. 
+
+URL: The url value is of type string and is not required, but when included should be a valid url 
+
+*/
+const referenceSchema = new mongoose.Schema({
+  text: {
+    type: String,
+    required: function () {
+      return this.url.length > 0;
+    },
+  },
+  url: {
+    type: String,
+    required: false,
+    validate(value) {
+      return /^(ftp|http|https):\/\/[^ "]+$/.test(value);
+    },
```

Comment:
use the match validation key instead.
```suggestion
    match: /^(ftp|http|https):\/\/[^ "]+$/,
```

Reply:
done